trade-finance-manager-api/src/v1/__mocks__/amendmentVariables.js

```diff
@@ -217,6 +218,30 @@ const noTaskVariables = {
   },
 };
 
+const dealWithNoCompanyName = {
+  ...firstTaskVariables,
+  dealSnapshot: {
+    ukefDealId,
+    exporter: {},
+    bankInternalRefName: 'UKEF',
+  },
+};
+
+const dealWithNoId = {
+  ...firstTaskVariables,
+  _id: null,
+};
+
+const noUkefDealId = {
+  ...firstTaskVariables,
+  dealSnapshot: {
+    exporter: {
+      companyName: 'Tester',
+    },
+    bankInternalRefName: 'UKEF',
+  },
+};
```

Comment:
This is somewhere that immer could be used to increase readability while keeping it all as one object --

```
const noUkefDealId = produce(firstTaskVariables, draft => {
    delete draft.dealSnapshot.ukefDealId;
});
```

But happy as is, as I think we should probably just look at replacing these with builders eventually and use this file as the director. But that's for another ticket/PR